compiler/AST/AstToText.cpp

```diff
@@ -1426,50 +1451,56 @@ void AstToText::appendExpr(IfExpr* expr, bool printingType)
 
 void AstToText::appendExpr(LoopExpr* expr, bool printingType)
 {
-  if (expr->forall)
+  std::string start,end;
+  if(expr->forall)
   {
-    if (expr->maybeArrayType)
+    if(expr->maybeArrayType)
     {
-      mText += '[';
-      if(expr->indices)
-      {
-        appendExpr(expr->indices, printingType);
-        mText += " in ";
-      }
-      
-      if(expr->iteratorExpr)
-      {
-        appendExpr(expr->iteratorExpr, printingType);
-        mText += ']';
+      start = "[";
+      end = "]";
+    }
 
-        if (BlockStmt* bs = toBlockStmt(expr->loopBody))
-        {
-          mText += ' ';
-          appendExpr(bs->body.get(1), printingType);
-        }
+    else
+    {
+      start = "forall ";
+      end = " do";
+    }
+  }
 
-        else
-        {
-          mText += "AppendExpr.Loop01";
-        }
-      
-      }
-      
-      else
-      {
-        mText += "AppendExpr.Loop02";
-      }
+  else
+  {
+    start = "for ";
+    end = " do";
+  }
+  mText += start;
+
+  if(expr->indices)
+  {
+    appendExpr(expr->indices, printingType);
+    mText += " in ";
+  }
+
+  if(expr->iteratorExpr)
+  {
+    appendExpr(expr->iteratorExpr, printingType);
+    mText += end;
+
+    if (BlockStmt* bs = toBlockStmt(expr->loopBody))
+    {
+      mText += ' ';
+      appendExpr(bs->body.get(1), printingType);
     }
 
     else
     {
-      mText += "AppendExpr.Loop03";
+      mText += "AppendExpr.Loop01";
```

Comment:
Hopefully won't be relevant, but it might be good to include a space at the start of this line, as otherwise it would output `...]AppendExpr.Loop01` or `... doAppendExpr.Loop01`